src/external.rs

```diff
@@ -0,0 +1,263 @@
+
+    let audio_data = if let Some((url, extension)) = audio_url {
+        if let Ok(res) = req_client.post(url).send().await {
+            Some((res.bytes(), extension))
+        } else {
+            None
+        }
+    } else {
+        None
+    };
+    let video_data = if let Some((url, extension)) = video_url {
+        if let Ok(res) = req_client.post(url).send().await {
+            Some((res.bytes(), extension))
+        } else {
+            None
+        }
+    } else {
+        None
+    };
+    let mut subs_data = Vec::with_capacity(subs_urls.len());
+    for (lang, _name, url, extension) in subs_urls {
+        if let Ok(res) = req_client.post(url).send().await {
+            subs_data.push((res.bytes(), lang, extension));
+        }
+    }
+    let thumbnail_data = if let Some(url) = thumbnail_url {
+        let extension = url
+            .rsplit('.')
+            .take(1)
+            .map(|s| s.to_owned())
+            .next()
+            .unwrap_or_default();
+        if let Ok(res) = req_client.post(url).send().await {
+            Some((res.bytes(), extension))
+        } else {
+            None
+        }
+    } else {
+        None
+    };
+
+    match (audio_data, video_data) {
+        (None, None) => panic![],
+        (None, Some(video_data)) => {
+            let video_pipe_path = format!["{base_path}/video"];
+            let output_path = format!["{base_path}/{title}.mkv"];
+            create(&video_pipe_path, Some(0o644)).unwrap();
+            let mut write_file = open_write(&video_pipe_path).unwrap();
+            write_file.write_all(&video_data.0.await.unwrap()).unwrap();
+            Command::new("ffmpeg")
+                .arg("-i")
+                .arg(video_pipe_path)
+                .arg("-r")
+                .arg("24")
+                .arg(output_path);
+        }
+        (Some(audio_data), None) => {
+            let audio_pipe_path = format!["{base_path}/audo"];
+            let output_path = format!["{base_path}/{title}.mka"];
+            create(&audio_pipe_path, Some(0o644)).unwrap();
+            let mut write_file = open_write(&audio_pipe_path).unwrap();
+            write_file.write_all(&audio_data.0.await.unwrap()).unwrap();
+            Command::new("ffmpeg")
+                .arg("-i")
+                .arg(audio_pipe_path)
+                .arg("-r")
+                .arg("24")
+                .arg(output_path);
+        }
+        (Some(_), Some(_)) => todo!(),
+    }
+}
+
+// (bv[height<=720]+ba/bv[height<=720]*+ba/b[height<=720])[filesize<100M]
+
+pub async fn download_info_json(url: Url) -> UrlHash {
+    let url_hash = hash_url(url.clone());
+    let path = format!["{STREAM_DIR}{}", url_hash.0];
+    std::fs::DirBuilder::new()
+        .recursive(true)
+        .create(&path)
+        .unwrap();
+    Command::new("yt-dlp")
+        .arg("-P")
+        .arg(path)
+        .args([
+            "--no-playlist",
+            "--write-info-json",
```

Comment:
You can use ` --dump-single-json` which outputs the whole json info to stdout instead

Reply:
Ended up doing this. The shorthand is `-j`.